include/Diagnostics.h

```diff
@@ -48,6 +48,7 @@ enum class DiagCode : uint8_t {
     BadDecimalDigit,
     BadHexDigit,
     DirectiveNotFirstOnLine,
+    TooManyErrors,
```

Comment:
Might want to call this TooManyLexerErrors so that it's obvious that it only applies to the lexer.